src/cats/SlideAnimation.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import clsx from "clsx";
+import styles from "./SlideAnimation.module.css";
+
+enum Direction {
```

Comment:
this is also an opinion, but I find that a simple string union is usually nicer than enums:

```
type Direction = 'previous' | 'next';
```

TypeScript is really good about the typing inside of switch, === comparison, and of course assignment.